### PR TITLE
Mirror incoming check

### PR DESCRIPTION
**Pull request by vladak**
Tested with a sample Mercurial and Git repo.

### CODE REVIEW

File: opengrok-tools/src/main/python/opengrok_tools/mirror.py [hunk 1]
```diff
@@ -43,30 +43,85 @@
...
+                              None,
+                              command_timeout)
+        if not repo:
+            logger.error("Cannot get repository for {}".
+                         format(repo_path))
+            ret = 1
+        else:
+            repos.append(repo)
+
+    return repos, ret
```
Comment (tulinkry) [inline opengrok-tools/src/main/python/opengrok_tools/mirror.py]: Consider using None for three-state return?

**Comment by vladak** [comment]
Also, the reason I selected `git update --dry-run` is that I wanted to avoid parsing the output of `git status` (in the case of `git fetch && git status -uno`).

**Comment by tulinkry** [comment]
I see `git pull --dry-run` not `update`

File: opengrok-tools/src/main/python/opengrok_tools/mirror.py [hunk 2]
```diff
@@ -352,14 +345,21 @@ def main():
...
-                # The error was already logged in get_repos_for_project()
+            repos = []
+            try:
+                global_config = {'commands': config.get(COMMANDS_PROPERTY),
+                                 'proxy': proxy,
+                                 'command_timeout': command_timeout,
+                                 'source_root': source_root,
+                                 'uri': uri}
+                repos = get_repos_for_project(logger, args.project,
+                                              ignored_repos, **global_config)
```
Comment (tulinkry) [inline opengrok-tools/src/main/python/opengrok_tools/mirror.py]: I think you can also write it like this

```python
get_repos_for_project(logger, args.project, ignored_repos,  commands=config.get(..), proxy=proxy, command_timeout=command_timeout, source_root=source_root, uri=uri)
```

choose the way you prefer.

**Comment by vladak** [comment]
Yep, it should've read `pull'.

File: opengrok-tools/src/main/python/opengrok_tools/mirror.py [hunk 3]
```diff
@@ -43,30 +43,78 @@
...
+                              # of repo_path is absolute path.
+                              kwargs['source_root'] + repo_path,
+                              repo_type,
+                              project,
+                              kwargs['commands'],
+                              kwargs['proxy'],
+                              None,
+                              kwargs['command_timeout'])
+        if not repo:
+            raise RepositoryException("Cannot get repository for {}".
```
Comment (tulinkry) [inline opengrok-tools/src/main/python/opengrok_tools/mirror.py]: +1 nice

**Pull request merged**